runtime/executor/src/main/java/edu/snu/coral/runtime/executor/data/block/FileBlock.java

```diff
@@ -120,10 +121,19 @@ private void writeSerializedPartitions(final Iterable<SerializedPartition<K>> se
         final K key = partitionMetadata.getKey();
         if (keyRange.includes(key)) {
           // The key value of this partition is in the range.
+          final long availableBefore = fileStream.available();
           final NonSerializedPartition<K> deserializePartition =
               DataUtil.deserializePartition(
                   partitionMetadata.getElementsTotal(), serializer, key, fileStream);
           deserializedPartitions.add(deserializePartition);
+          // rearrange file pointer
+          final long toSkip = partitionMetadata.getPartitionSize() - availableBefore + fileStream.available();
+          if (toSkip > 0) {
+            skipBytes(fileStream, toSkip);
+          } else if (toSkip < 0) {
+            final FileChannel fc = fileStream.getChannel();
+            fc.position(fc.position() + toSkip);
```

Comment:
We should avoid to use `FileChannel#position` method. We've experienced it's malfunction for a while.

Reply:
@sanha Fixed. Thank you!